diff -ur a/sipXtackLib/src/net/SipMessage.cpp b/sipXtackLib/src/net/SipMessage.cpp
--- a/sipXtackLib/src/net/SipMessage.cpp
+++ b/sipXtackLib/src/net/SipMessage.cpp
@@ -3806,13 +3806,13 @@
             routeString.append(SIP_MULTIFIELD_SEPARATOR);
         }
         // Make sure the route is in name-addr format
-        if(strstr(routeUri,"<") <= 0)
+        if(strstr(routeUri,"<") == nullptr)
         {
             routeString.append("<");
         }
 
         routeString.append(routeUri);
-        if(strstr(routeUri, ">") <= 0)
+        if(strstr(routeUri, ">") == nullptr)
         {
             routeString.append(">");
         }
